runtime/cudaq/platform/default/rest/helpers/anyon/AnyonServerHelper.cpp

```diff
@@ -0,0 +1,464 @@
+      {"Accept", "*/*"}};
+  return headers;
+}
+
+RestHeaders AnyonServerHelper::getHeaders() { return generateRequestHeader(); }
+
+/// Refresh the api key and refresh-token
+void AnyonServerHelper::refreshTokens(bool force_refresh) {
+  std::mutex m;
+  std::lock_guard<std::mutex> l(m);
+  RestClient client;
+  auto now = std::chrono::high_resolution_clock::now();
+
+  if (apiKey.empty()) {
+    force_refresh = true;
+    if (refreshKey.empty())
+      refreshKey = credentials;
+  }
+  if (timeStr.empty()) {
+    timeStr = std::to_string(now.time_since_epoch().count());
+  }
+
+  // We first check how much time has elapsed since the
+  // existing refresh key was created
+  std::int64_t timeAsLong = std::stol(timeStr);
+  std::chrono::high_resolution_clock::duration d(timeAsLong);
+  std::chrono::high_resolution_clock::time_point oldTime(d);
+  auto secondsDuration =
+      1e-3 *
+      std::chrono::duration_cast<std::chrono::milliseconds>(now - oldTime);
+
+  // If we are getting close to an 30 min, then we will refresh
+  bool needsRefresh = secondsDuration.count() * (1. / 1800.) > .85;
+  if (needsRefresh || force_refresh) {
+    cudaq::info("Refreshing id_token");
+    std::stringstream ss;
+    ss << "\"refresh_token\":\"" << refreshKey << "\"";
+    auto headers = generateRequestHeader(refreshKey);
+    nlohmann::json j;
+    j["refresh_token"] = refreshKey;
+    auto response_json = client.post(baseUrl, "login", j, headers);
+    apiKey = response_json["id_token"].get<std::string>();
+    refreshKey = response_json["refresh_token"].get<std::string>();
+    std::ofstream out(credentialsPath);
+    out << "key:" << apiKey << '\n';
+    out << "refresh:" << refreshKey << '\n';
+    out << "time:" << now.time_since_epoch().count() << '\n';
+    timeStr = std::to_string(now.time_since_epoch().count());
+  }
+  // If the time string is empty, let's add it
+  if (timeStr.empty()) {
+    timeStr = std::to_string(now.time_since_epoch().count());
+    std::ofstream out(credentialsPath);
+    out << "key:" << apiKey << '\n';
+    out << "refresh:" << refreshKey << '\n';
+    out << "time:" << timeStr << '\n';
+  }
+}
+
+void findApiKeyInFile(std::string &apiKey, const std::string &path,
+                      std::string &refreshKey, std::string &timeStr,
+                      std::string &credentials) {
+  std::ifstream stream(path);
+  std::string contents((std::istreambuf_iterator<char>(stream)),
+                       std::istreambuf_iterator<char>());
+
+  std::vector<std::string> lines;
+  lines = cudaq::split(contents, '\n');
+  nlohmann::json jsoncreds;
+  for (const std::string &l : lines) {
+    std::vector<std::string> keyAndValue = cudaq::split(l, ':');
+    // if (keyAndValue.size() != 2)
+    //   throw std::runtime_error("Ill-formed configuration file (" + path +
+    //                            "). Key-value pairs must be in `<key> : "
+    //                            "<value>` format. (One per line)");
```

Comment:
Is there any particular reason this is commented out? If not, could you please uncomment it?

Reply:
We assumed that a user could create an initial configuration file for logging in with "credentials: {username: <username>, password:<password>}". Splitting the line by ":" will lead to three segments. Uncommenting this line will throw an error when it shouldn't. Upon successful initial authentication, the configuration fille will be updated with an api key and a refresh key. Future access and authentication will be done through the tokens. Probably to make it less confusing and restore a guard on the contents of the credential configuration, we could change the condition to "(keyAndValue.size() != 2) &&( ((keyAndValue[0] != "credentials") || (keyAndValue.size() != 3))"

Reply:
just implemented the change